### PR TITLE
Display active banner after new user session

### LINKED ISSUE #5428: Bug: Banner dismissed forever
## Impacted User Types
 - volunteers
 - supervisors
 - admins

## Environment
QA and Prod

## Current Behavior
When a user dismisses an "active" banner in one session, the banner never shows up for them again - it is dismissed forever

https://www.loom.com/share/9d0312d7095b4419bf760bcd0de22d0f

## Expected Behavior
active banners should show up for every session for a user as long as the banner is "active"

## How to Replicate
ex: 
1. Log in as an admin or supervisor.
2. Click on "Banners"
3. Create a new banner and set it to "active"
4. Dismiss the banner for yourself
5. End your session by logging out and log back in
6. Notice the banner is not showing for you despite it still being active

## How to access the QA site
_Login Details:_  
[Link to QA site](https://casa-qa.herokuapp.com/)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin
- all_casa_admin1@example.com view site as an all casa admin
  - go to `/all_casa_admins/sign_in`  

password for all users: 12345678  

### Questions? Join Slack!

We highly recommend that you join us in [slack](http://bit.ly/3Quxc1Q) #casa channel to ask questions quickly. And [discord](https://discord.gg/qJcw2RZH8Q) for office hours (currently Tuesday 5-7pm Pacific), stakeholder news, and upcoming new issues.

### PR DESCRIPTION
**Pull request by gorangorun**
### What github issue is this PR for, if any?
Resolves #5428

### What changed, and why?
How it works: After clicking dismiss banner, cookie is set, meaning that banner should not be displayed. After signout, cookie persisted, thus on next login banner is still hidden.

How it should work after this PR: After clicking dismiss banner, cookie is set. Clicking signout will wipe out the cookie resulting in displaying the banner on next login.

### How will this affect user permissions?
Not related to user permissions

### How is this tested? (please write tests!) 💖💪
Tests included (spec/helpers/banner_helper_spec.rb)

### Feelings gif (optional)
What gif best describes your feeling working on this issue? https://giphy.com/
How to embed:
`alt text`

### Feedback please? (optional)
We are very interested in your feedback! Please give us some :) https://forms.gle/1D5ACNgTs2u9gSdh9

### CODE REVIEW

**Comment by h-m-m** [comment]
Interesting. I was considering moving the alert state from a separate cookie to the Rails session. Doing so might free us from clearing the state on logout, but would require moving the storage from JS setting and unsetting cookies to JS making a round-trip to the server. I wasn't sure which was the better route.

Since you've written this out, I'm fine with going this way

File: app/views/layouts/_banner.html.erb [hunk 1]
```diff
@@ -1,14 +1,10 @@
-<% banner = current_organization.banners.active.first %>
-<% if banner %>
-  <% cookie_id = "banner_#{banner.id}" %>
-  <% if !cookies["dismiss_#{cookie_id}"] %>
```
Comment (h-m-m) [inline app/views/layouts/_banner.html.erb:4]: I like that you've removed this logic from the view

**Comment by h-m-m** [comment]
Should we add a test that asserts the cookie gets cleared when the SignoutCleanupController is invoked?

**Comment by gorangorun** [comment]
@h-m-m well, i didnt found any tests for other controllers, so i thought it's not needed. 

@FireLemons have time to review this?

**Comment by schoork** [comment]
@gorangorun Definitely want tests for:
1.  the banner is dismissed when the user clicks the dismiss button
2. the banner shows when the user logs back in

**Comment by schoork** [comment]
@gorangorun Would changing from using cookies to session storage in the original implementation fix the problem?

**Comment by bcastillo32** [comment]
> @gorangorun Would changing from using cookies to session storage in the original implementation fix the problem?

Hi @gorangorun we hope you had a great holiday and break. please let us know if you could use some help on this. @schoork you mentioned we need some test for this, yes?

**Comment by schoork** [comment]
> > @gorangorun Would changing from using cookies to session storage in the original implementation fix the problem?
> 
> Hi @gorangorun we hope you had a great holiday and break. please let us know if you could use some help on this. @schoork you mentioned we need some test for this, yes?

Yeah, definitely want some tests for this @bcastillo32. I can take a look this weekend.

**Comment by schoork** [comment]
@bcastillo32 PR #5473 is an alternate PR with tests. If that one is merged, this one should be closed without merging.

**Comment by compwron** [comment]
> @bcastillo32 PR #5473 is an alternate PR with tests. If that one is merged, this one should be closed without merging.

https://github.com/rubyforgood/casa/pull/5473 was merged, so I am closing this PR

**Pull request closed without merging**